app/controllers/oidc_controller.rb

```diff
@@ -111,7 +111,7 @@ def client
       authorization_endpoint: config.authorization_endpoint,
       token_endpoint:         config.token_endpoint,
       userinfo_endpoint:      config.userinfo_endpoint,
-      redirect_uri:           url_for(controller: "oidc", action: "done")
+      redirect_uri:           oidc_done_url(host: request.base_url)
```

Comment:
Isn't this change effectively a noop? I would assume `oidc_done_url()` under the covers does a `url_for(controller: "oidc", action: "done")`, and that `host` defaults to `request.base_url`?

Ideally, we should redirect to `request.base_url` IF `request.base_url  matches to either the dashboard internal fqdn (i.e. the `dashboard` pillar value) or the dashboard external fqdn (i.e. the `dashboard_external_fqdn` pillar value), If `request.host` does not match either of these, then we should redirect to `dashboard_external_fqdn`.

The reasoning here - is Dex will only accept those 2 names (as those are the 2 names we gave it). Anything else will be rejected, so we should try and use whichever name the user has browsed to - and fallback to the external_fqdn

Reply:
I agree, we should set `host` to whatever that was (if it's on the "whitelist" host -- also accepted by dex), otherwise fallback to `Pillar.value(:dashboard_external_fqdn)`.

The only problem I see with this approach and that I never tested is what happens if the session is missing (if we get to the fallback, the Velum session will probably not exist for `dashboard_external_fqdn` host), and I don't know if we will be redirecting automatically to whatever path was requested.

Reply:
i just tested this, and indeed the session gets invalid. so the user always needs to access from either the host of the `dashboard` or `dashboard_external_fqdn` pillar value.

i already tried to switch from sessions to rails cache but that didnt work that reliable (also in dev we have cache disabled)

should we add this as a warning in the branding, or should we log out the user and send a flash message with the information that it needs to be accessed via the registered one of the 2 fqdn's? or do you have any other idea that's better?